packages/sdk/src/client.ts

```diff
@@ -728,6 +730,57 @@ export class Client
         return { streamId: streamId, prevMiniblockHash: streamView.prevMiniblockHash }
     }
 
+    async createSpaceMediaStream(
+        spaceId: string | undefined,
+        chunkCount: number,
+        streamSettings?: PlainMessage<StreamSettings>,
+    ): Promise<{ streamId: string; prevMiniblockHash: Uint8Array }> {
+        assert(spaceId !== undefined, 'userStreamId must be set')
+        assert(this.userStreamId !== undefined, 'userStreamId must be set')
+        assert(isSpaceStreamId(spaceId), 'spaceId must be a valid streamId')
+
+        const streamId = makeMediaStreamIdFromSpaceId(spaceId)
+        const streamIdBytes = streamIdAsBytes(streamId)
+        const spaceIdBytes = streamIdAsBytes(spaceId)
+
+        this.logCall('createSpaceMediaStream', streamId)
+
+        const inceptionEvent = await makeEvent(
+            this.signerContext,
+            make_MediaPayload_Inception({
+                streamId: streamIdBytes,
+                spaceId: spaceIdBytes,
+                channelId: spaceIdBytes,
+                chunkCount,
+                settings: streamSettings,
+                publicScope: PublicScope.PS_SPACE,
+            }),
+        )
+
+        const response = await this.rpcClient.createStream({
+            events: [inceptionEvent],
+            streamId: streamIdBytes,
+        })
+
+        const unpackedResponse = await unpackStream(response.stream)
+        const streamView = new StreamStateView(this.userId, streamId)
+        streamView.initialize(
+            unpackedResponse.streamAndCookie.nextSyncCookie,
+            unpackedResponse.streamAndCookie.events,
+            unpackedResponse.snapshot,
+            unpackedResponse.streamAndCookie.miniblocks,
+            [],
+            unpackedResponse.prevSnapshotMiniblockNum,
+            undefined,
+            [],
+            undefined,
+        )
+
+        check(isDefined(streamView.prevMiniblockHash), 'prevMiniblockHash must be defined')
+
+        return { streamId: streamId, prevMiniblockHash: streamView.prevMiniblockHash }
```

Comment:
Are you sure we need a new function here? Can we just make channel id optional and add the scope param to the existing makeMediaStream function?

Reply:
There are at least 3 differences:
1. The assert rules are different between a regular media stream and a public media stream
2. The inception payloads are different
3. A 'public decryption key' will be generated and made available. Doing this in a follow up PR.

I had a first cut with `if..else`, but it looked a bit messy to me. So I separated them. I can re-do it in if..else to see how it reads.

Reply:
fixed